.jenkins/pipelines/Azure/Linux/Jenkinsfile

```diff
@@ -18,21 +18,86 @@ AGENTS_LABELS = [
     "acc-ubuntu-18.04-vanilla": env.UBUNTU_VANILLA_1804_CUSTOM_LABEL ?: "vanilla-ubuntu-1804"
 ]
 
+def CmakeArgs(String build_type = "RelWithDebInfo", String code_coverage = "OFF", String debug_malloc = "ON", String lvi_args="", String cmake_args = "") {
+    def args = "-G Ninja -DCMAKE_INSTALL_PREFIX:PATH='/opt/openenclave' -DCPACK_GENERATOR=DEB -DCODE_COVERAGE=${code_coverage} -DUSE_DEBUG_MALLOC=${debug_malloc} -DCMAKE_BUILD_TYPE=${build_type} ${lvi_args} ${cmake_args} -Wdev"
+    return args
+}
+
+def TestCommand() {
+    def buildAndTest = """
+        echo "Running Test Command"
+        ctest --output-on-failure --timeout ${CTEST_TIMEOUT_SECONDS}
+    """
+    return buildAndTest
+}
+
+def InstallBuildCommand() {
+    def localPackage = """
+        echo "Running Install Build Command"
+        cpack -G DEB
+        sudo killall -9 apt && sudo killall -9 dpkg
+        sudo ninja -v install
+    """
+    return localPackage
+}
+
+def InstallReleaseCommand() {
```

Comment:
This seems to only be used for the upgrade tests, so defining it outside of that function seems to be moot for now. Do you think this will be used by any other test types? If not, it would probably be cleaner to move this directly into upgrade test.

Reply:
I don't think it'll be used for anything else right now, but a secondary goal with moving the commands into functions was to clean up the test declaration. In my opinion it was easier to understand what the tests were trying to accomplish by reading the function calls instead of big blocks of text.